Fix raycasts hitting gizmos + Fix Copy Component Implementation

989onan opened the pull request:
the implementation is kinda bad, but I managed to transpile it in so the entire if statement can be manipulated.

989onan committed:
Animator no longer broke

I took personal offense that my mantis blade that was driven didn't work with
this mod, so I fixed it.

989onan committed:
Merge branch 'master' into master

989onan committed:
Merge remote-tracking branch 'upstream/master'

989onan committed:
stap

989onan committed:
well I tried

this does what I asked it to do (first try actually!)
But it doesn't do what I want.

I just want objects when being grabbed not yoinking users into them if they have character parenters. I am too lazy to make a github issue it takes YDMS ages to fix litterally anything like this.

989onan committed:
Merge branch 'ResoniteModdingGroup:master' into master

989onan committed:
fix raycasts hitting gizmos

I don't like the implementation, but it will do

989onan committed:
Merge branch 'master' of https://github.com/989onan/ResoniteCommunityBugFixCollection

989onan committed:
Revert "Merge branch 'master' of https://github.com/989onan/ResoniteCommunityBugFixCollection"

This reverts commit 0ec1919e50fc78784fbadc6a66abf94b3537f77d, reversing
changes made to 753d706a03ce9ef337380e8b1fefe2da66eaa15c.

Nytra commented:
Unnecessary recomputation of ToString

Banane9 commented:
> Unnecessary recomputation of ToString

I'm not even sure what that's doing - checking for which field or such something loads?

That should be done through getting the FieldInfo / MethodInfo and then using the `Loads` or `Calls` extensions.

989onan commented:
>  I wanted to make it to where if a line is changed, a constant \"74\" and \"95\" wouldn't break if frooxius added code.

These are the actual IL instructions in the common update method at the time of this PR. Essentially encompasses the if statement minus the """branch if false""" which I keep when injecting the call to my method, as to not break the if statement's jump to next IL instruction block when mine returns false to not hit the collider 

.(Unrelated, but this patch saved me so much headache yesterday when doing a RaycastDriver for someone last night)

989onan committed:
rebase and fix

basically add funny that tries to access the methods. this is untested

989onan committed:
Merge branch 'master' into master

989onan committed:
Merge pull request #1 from ResoniteModdingGroup/master

rebase

989onan committed:
I said work dammit

I want this to work, not be a pain in my ass, not complex with some stupid access tools, just feking work

989onan committed:
Update BreakDragAndDropCopiedComponentDrives.cs

- make it a transpiler instead of a full code replacement for one line, so it doesn't conflict violently with other mods
- this method generates bad IL code. Why? Has I ever? I hate modding sometimes and I'm too dumb to fix it

989onan committed:
Others fix my sanity

I got it figured out thanks to Nytra, Cyro, Art0007i and others. This should work now. X_X

989onan commented:
now two required things for my workflow that I have to keep compiling on top of new changes without this PR approved. Wonderful...

Banane9 committed:
Merge branch 'master' into master
